rotkehlchen/chain/ethereum/adex/adex.py

```diff
@@ -0,0 +1,1034 @@
+            nonce: int,
+    ) -> HexStr:
+        """Given a LogBond event data, return its `bondId`.
+        """
+        arg_types = ['address', 'address', 'uint', 'bytes32', 'uint']
+        args = [STAKING_ADDR, identity_address, amount, pool_id, nonce]
+        return HexStr(Web3.keccak(Web3().codec.encode_abi(arg_types, args)).hex())
+
+    def _get_identity_address_map(
+            self,
+            addresses: List[ChecksumEthAddress],
+    ) -> Dict[ChecksumAddress, ChecksumAddress]:
+        """Returns a map between the user identity address in the protocol and
+        the EOA/contract address.
+        """
+        return {self._get_user_identity(address): address for address in addresses}
+
+    def _get_tom_pool_unclaimed_rewards(
+            self,
+            addresses: List[ChecksumEthAddress],
+    ) -> Dict[ChecksumAddress, UnclaimedReward]:
+        """Get Tom pool unclaimed rewards per address
+
+        NB: the unclaimed rewards are ADX and DAI balances associated with the
+        user identity of the address.
+        """
+        fee_rewards = self._get_tom_pool_fee_rewards_from_api()
+        identity_address_map = self._get_identity_address_map(addresses)
+        address_adx_amount = defaultdict(list)
+        address_dai_amount = defaultdict(list)
+        for entry in fee_rewards:
+            try:
+                token_ethereum_address = entry['channelArgs']['tokenAddr']
+                balances = entry['balances']
+                if token_ethereum_address == A_ADX.ethereum_address:
+                    for user_identity in identity_address_map.keys():
+                        address = identity_address_map[user_identity]
+                        address_adx_amount[address].append(FVal(balances.get(user_identity, ZERO)))
+                elif token_ethereum_address == A_DAI.ethereum_address:
+                    for user_identity in identity_address_map.keys():
+                        address = identity_address_map[user_identity]
+                        address_dai_amount[address].append(FVal(balances.get(user_identity, ZERO)))
+                else:
+                    log.error(
+                        'Unexpected token address processing AdEx unclaimed rewards. ',
+                        token_address=token_ethereum_address,
+                    )
+            except (KeyError, ValueError) as e:
+                msg = str(e)
+                log.error(
+                    'Error processing AdEx Tom pool unclaimed rewards. ',
+                    entry=entry,
+                    error=msg,
+                )
+                self.msg_aggregator.add_error(
+                    'Failed to deserialize the AdEx Tom pool unclaimed rewards. '
+                    'Check logs for details. Ignoring it.',
+                )
+                return {}
+
+        unclaimed_rewards = {}
+        for address in addresses:
+            total_adx_amount = FVal(sum(address_adx_amount[address]))
```

Comment:
you don't need to FVal something that is already definitely an FVal. This is not simple casting. It's a constructor, so calling it unnecessarily impacts performance.

Reply:
Fixed it. Please resolve when you read it.